test/libyul/yulOptimizerTests/loadResolver/memory_with_extcall_invalidation.yul

```diff
@@ -0,0 +1,22 @@
+{
+    mstore(2, 9)
+    sstore(0, mload(2))
+    pop(extcall(0, 0, 0, 0))
+    sstore(0, mload(2))
+}
+// ====
+// bytecodeFormat: >=EOFv1
+// ----
+// step: loadResolver
+//
+// {
+//     {
+//         let _1 := 9
+//         mstore(2, _1)
+//         let _4 := _1
+//         let _5 := 0
+//         sstore(_5, _4)
+//         pop(extcall(_5, _5, _5, _5))
+//         sstore(_5, _1)
```

Comment:
Nice. The original test had an unresolved `mload()` here.

This seems to be the practical effect of the fact that we have to assume that `CALL` writes memory but [`EXTCALL` can only ever read it](https://github.com/ethereum/solidity/blob/f632ec84efa2bf896934d5c9466ea79d9db4e142/libevmasm/SemanticInformation.cpp#L501).


`CALL`'s write could be figured out to have no impact but [apparently we ignore its length](https://github.com/ethereum/solidity/blob/f632ec84efa2bf896934d5c9466ea79d9db4e142/libevmasm/SemanticInformation.cpp#L161-L169).

Reply:
But I just noticed something in `SemanticInfo.cpp` that looks like a bug: https://github.com/ethereum/solidity/blob/f632ec84efa2bf896934d5c9466ea79d9db4e142/libevmasm/SemanticInformation.cpp#L184-L188

Looks like we decided to mark `EXTDELEGATECALL` as writing storage/tstorage, but not `EXTCALL`, even though `CALL` is marked as such. I may have been thinking only about direct writes when I reviewed it but I guess `EXTCALL` can modify storage indirectly, by calling a function that can do that. We should fix that (and make sure there's a test where the difference shows).